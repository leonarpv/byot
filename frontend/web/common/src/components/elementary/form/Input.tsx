import React from 'react';
import {makeStyles, TextField, TextFieldProps, Theme} from '@material-ui/core';
import {CreateCSSProperties, CSSProperties} from '@material-ui/styles';
import classNames from 'classnames';
import {WithStyles} from '../../../types/WithStyles';
import {FormHelperText} from './FormHelperText';
import {debounce} from 'lodash';

export type Props = Omit<TextFieldProps, 'variant'> &
  WithStyles<typeof styles> & {
    debounce?: number;
  };

const styles = (theme: Theme) => ({
  filledRoot: (props: Props) =>
    ({
      borderRadius: theme.shape.borderRadius,
      backgroundColor: theme.palette.background.paper,
      '&:hover': !props.disabled && {
        backgroundColor: theme.palette.type == 'light' && theme.palette.grey[200],
      },
      opacity: props.disabled ? 0.1 : 1,
    } as CreateCSSProperties<Props>),
  label: {
    color: theme.palette.grey[500],
  },
});
const useStyles = makeStyles(styles);

export const Input: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  const {helperText, ...p} = props;
  const onChange: React.ChangeEventHandler<HTMLInputElement> | undefined =
    (props.debounce || 0) > 0 && props.onChange ? debounce(props.onChange, props.debounce) : props.onChange;
  return (
    <>
      <TextField
        fullWidth
        {...p}
        onChange={onChange}
        variant="filled"
        InputLabelProps={{
          ...props.InputLabelProps,
          classes: {
            root: classNames(styles.label, props.InputLabelProps?.classes?.root),
          },
        }}
        InputProps={{
          disableUnderline: true,
          ...props.InputProps,
          classes: {
            root: classNames(styles.filledRoot, props.InputProps?.classes?.root),
          },
        }}
      />
      <FormHelperText error={props.error} variant="filled">
        {props.helperText}
      </FormHelperText>
    </>
  );
};
