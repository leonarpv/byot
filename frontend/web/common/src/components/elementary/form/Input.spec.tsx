import * as React from 'react';
import {render} from '@testing-library/react';
import {Input} from './Input';

describe('<Input/>', () => {
  it('should render', () => {
    const {container} = render(<Input />);
    expect(container).toMatchSnapshot();
  });
});
