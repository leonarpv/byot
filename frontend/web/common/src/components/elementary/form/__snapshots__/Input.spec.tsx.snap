// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should render 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-1 makeStyles-filledRoot-3 MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="MuiCollapse-container MuiCollapse-hidden"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner"
      >
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
          <span>
            â€‹
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
